components/AlertNotification/AlertNotification.test.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Alert from './AlertNotification';
+import { Variant } from './Alerts.enum';
+import { CheckCircle, XCircle, Info, AlertTriangle } from 'lucide-react';
+
+const variantTestCases = [
+  {
+    variant: Variant.Success,
+    title: 'Success!',
+    message: 'This is a success message',
+    icon: CheckCircle,
+  },
+  {
+    variant: Variant.Error,
+    title: 'Error!',
+    message: 'This is an error message',
+    icon: XCircle,
+  },
+  {
+    variant: Variant.Default,
+    title: 'Info',
+    message: 'This is an info message',
+    icon: Info,
+  },
+  {
+    variant: Variant.Warning,
+    title: 'Warning!',
+    message: 'This is a warning message',
+    icon: AlertTriangle,
+  },
+];
+
+describe('AlertNotification', () => {
+  variantTestCases.forEach(({ variant, title, message }) => {
+    it(`renders the correct variant ${variant}`, () => {
+      render(<Alert variant={variant} message={message} />);
+
+      if (variant) {
+        expect(screen.getByText(title)).toBeInTheDocument();
```

Comment:
Right now I don't see that your test is checking for the proper title, message, nor it looking at the icon at all. Perhaps you can write an if statement to check for the appropriate content?

For example:

```tsx
if (variant = Success) {
expect(screen.AlertTitle).toHaveTextContent('Success!')
}
``` 

and so on.

Reply:
You shouldn't need a condition. The for each will loop through the content, and it should render the object's content.

Reply:
do you mean that I only need to write three lines of code to basically just check that title, message, and an icon display?

For example:
```tsx
expect(screen.getByTestId('alert-title').toHaveTextContent('something')
expect(screen.getByTestId('alert-message').toHaveTextContent('Something')
expect(screen.getByTestId('alert-icon')).toBeInTheDocument();
```
and that's it? or to check that each variant displays the right info?